syntax = "proto3";

option go_package = "./;modelx";

package model;

// The greeting service definition.
service Modelx {
  rpc Ping (PingRequest) returns (PongReply) {}

  // generate embedding vector for passed sentence.
  rpc GenEmbedding(EmbeddingRequest) returns (EmbeddingReply) {}

  // bulk generate embedding vectors.
  rpc GenEmbeddingList(EmbeddingListRequest) returns (EmbeddingListReply) {}

  // calc similarity score for sentence.
  rpc CalcSimilarityScore(SimilarityRequest) returns (SimilarityReply) {}
}

message PingRequest {
}

message PongReply {
  int32 code = 1;
}

message EmbeddingRequest {
  string text = 1;
}

message EmbeddingReply {
  string embedding = 1;
}

message EmbeddingListRequest {
  repeated string text_list = 1;
}

message EmbeddingListReply {
  repeated string embedding_list = 1;
}

message SimilarityRequest {
  string source_text = 1;
  repeated string target_texts = 2;
}

message SimilarityReply {
  repeated double scores = 1;
}

